src/components/Calculator.js

```diff
@@ -1,49 +1,46 @@
-// eslint disable no-named-as-default;
-import React from 'react';
+import React, { useState, setState } from 'react';
 import MagicBtn from './MagicBtn';
 import OutputResult from './OutputResult';
 import calculate from '../logic/calculate';
+import generateOutputValue from '../helper/generateOutputvalue';
 
-const generateOutputValue = ({ total, next, operation }) => {
-  if (!total && !next && !operation) {
-    return '0';
-  } if (!total && next) {
-    return `${next}`;
-  } if (operation && total && !next) {
-    return `${total} ${operation}`;
-  } if (total && next && operation) {
-    return `${total} ${operation} ${next}`;
-  }
-  return `${total}`;
-};
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
+const Calculator = () => {
+  const [calculateState, calculateStateHandler] = useState(
+    {
+      total: null,
+      next: null,
+      operation: null,
+    },
+  );
 
-  clickEventListener = (event) => {
-    const object = this.state;
+  const clickEventListener = (event) => {
     const { value } = event.target;
-    const result = calculate(object, value);
-    this.setState(result);
-  }
+    let { total, next, operation } = calculate(calculateState, value);
+    if (total === undefined) {
+      total = setState(calculateState.total);
+    }
+    if (next === undefined) {
+      next = setState(calculateState.next);
+    }
+    if (operation === undefined) {
+      operation = calculateState.operation;
+    }
+    calculateStateHandler({ total, next, operation });
+  };
 
-  render() {
-    const outputValue = generateOutputValue(this.state);
-    const buttonKeys = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '1', '='];
-    const buttons = buttonKeys.map((val) => (
-      <MagicBtn value={val} key={0} clickEventListener={this.clickEventListener} />
-    ));
-    return (
-      <div className="magician">
-        <OutputResult outputvalue={outputValue} />
-        <div className="magician-keys">
-          {buttons}
-        </div>
+  const outputValue = generateOutputValue(calculateState);
+  const buttonKeys = ['AC', '+/-', '%', '÷', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '1', '='];
+  const buttons = buttonKeys.map((val) => (
```

Comment:
- [ ] I think you forgot to add `.` button between the button `0` and `=` you used `1` twice check that and fix it

Reply:
### Hi @DeliceLydia ❤️‍🔥 
- I certainly was pissed at first that such a tiny mistake could stop my progress, But all then I realise that your attention to detail is top-notch 🤝 . 
- Please even dough I really admire what you did sincerely I would have been happier 😍  if you had shown more empathy by reaching out first or point it as [optional] changes rather than not approving because of that little tiny bug.
- Really put in a lot that I forgot to rest well and end up smuggling that bug it was an honest mistake I am fixing it right away and hey 😉 !
#### Thanks anyway I am still 😎  😄